jurity/recommenders/rank_estimation.py

```diff
@@ -0,0 +1,47 @@
+from typing import NamedTuple
+
+import numpy as np
+import pandas as pd
+
+from jurity.utils import Constants, get_sorted_clicks
+
+
+class RankEstimation:
+    def __init__(self, n_items: int = 1, n_sampled: int = 1):
+        """ Unbiased Rank Estimation
+
+        Performs a correction of ranking metrics for use cases where the negative responses were sampled.
+        This adjusts the metric for when if the full set of items were used,
+        items in the sampled top k would not be ranked as high.
+
+        .. math::
+            URE = metric \\times (1 + \\frac{(n_sampled -1)(predicted_rank - 1)}{n_items})
+
+        Parameters
+        ----------
+        n_items: int
```

Comment:
Where do these parameters come from? Is `n_items` all items and `n_sampled` is the number of items included in the response matrix used for training?

Reply:
This is only necessary if you sampled the test data. n_items is the total number of items (and in writing that I'm realizing this may need to allow an array for accommodating eligibility rules) and n_sampled is the number that was sampled for the test data set. So for example when we were assuming what was used in Viewpoints based on what the top 10 articles were for that day, n_sampled would be 10